code/SingleRecordAdmin.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace SilverStripe\Admin;
+
+use SilverStripe\Forms\Form;
+use SilverStripe\ORM\DataObject;
+
+/**
+ * The base class for admin sections that exist for editing a single record.
+ */
+abstract class SingleRecordAdmin extends LeftAndMain
+{
+    /**
+     * Determines if there should be a single record in the database that this admin edits.
+     *
+     * If this is not true, you need to provide a mechanism to tell the form which record it should be editing.
+     * This could be an action (e.g. edit/$ID), or could be based on something about the current member, etc.
+     */
+    private static bool $only_one_record = true;
```

Comment:
This is for `SiteConfig` style functionality - but since `CMSProfileController` is explicitly for the _current_ member and not _the one member in the database_, it needs to be configurable.